script.js

```diff
@@ -0,0 +1,70 @@
+const currencyRates = [];
+
+function addCurrenyObject(fromCurrency, toCurrency, rate, date, timestamp) {
+  const currencyObj = {
+    timestamp: timestamp,
+    base: fromCurrency,
+    date: date,
+    toCurrency: toCurrency,
+    rates: {},
+    addRate: function () {
+      if (rate != undefined) {
+        this.rates[toCurrency] = rate;
+      }
+    },
+  };
+
+  currencyRates.push(currencyObj);
+  currencyObj.addRate();
+
+  console.log(currencyObj);
+}
+
+function addRate(fromCurrency, toCurrency, rate, date, timestamp) {
+  for (let item in currencyRates) {
+    if (item.base === fromCurrency) {
+      item.addRate(toCurrency, rate);
```

Comment:
Your `for...in` loop is iterating incorrectly over `currencyRates`. for...in loops over keys, not objects. You can use `forEach` or `find( )` instead